packages/cli/src/build.test.ts

```diff
@@ -1,951 +1,394 @@
+          'dist/index.d.cts',
+        ]);
+      });
     });
-
-    expect(output).toMatchInlineSnapshot(`
-      ""use strict";
-      // no-op
-      "
-    `);
   });
 
   it('removes the output directory if `clean` is enabled', () => {
-    const environment = getVirtualEnvironment({
-      files: {
-        '/index.ts': '// no-op',
-      },
-      tsconfig: {
-        compilerOptions: {
-          outDir: '/non-existent',
-        },
-      },
-    });
+    const path = getFixture('node-16');
+    const dist = join(path, 'dist');
 
-    compile({
-      format: 'module',
-      environment: {
-        ...environment,
-        system: {
-          ...environment.system,
-          createDirectory: noOp,
-        },
-      },
+    compile(path, ['module'], {
       clean: true,
-      outputPath: '/non-existent/index',
     });
 
-    expect(removeDirectory).toHaveBeenCalledWith('/non-existent', '/');
+    expect(vi.mocked(removeDirectory)).toHaveBeenCalledWith(dist, path);
   });
 
   it('throws an error if the project fails to initialise', () => {
-    const environment = getVirtualEnvironment({
-      files: {
-        '/index.ts': 'import { foo } from "./foo";',
-      },
-      checkDiagnostic: false,
-    });
-
-    expect(() =>
-      compile({
-        format: 'module',
-        environment,
-      }),
-    ).toThrow(
-      "Cannot find module './foo' or its corresponding type declarations.",
+    expect(() => compile(getFixture('invalid'), ['module'])).toThrow(
+      'Failed to initialise the project.',
     );
   });
 
-  it('throws an error if the project fails to build', () => {
-    const { program, system } = getVirtualEnvironment({
-      files: {
-        '/index.ts': 'import { foo } from "./foo";',
+  it('logs an error if the project fails to build', () => {
+    const error = vi.spyOn(console, 'error').mockImplementation(noOp);
+    const system: System = {
+      ...sys,
+      writeFile() {
+        throw new Error('Failed to write file.');
       },
-      checkDiagnostic: false,
-    });
+    };
 
-    program.emit = () => ({
-      emitSkipped: true,
-      diagnostics: [
-        {
-          category: 1,
-          code: 2307,
-          messageText: "Cannot find module './foo'.",
-          file: undefined,
-          start: undefined,
-          length: undefined,
-        },
-      ],
+    buildHandler({
+      // ESLint doesn't seem to be able to infer the type of `getFixture` here.
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      project: `${getFixture('node-16')}/tsconfig.json`,
+      format: ['module'],
+      clean: false,
+      system,
     });
 
-    expect(() =>
-      build({
-        program,
-        system,
-        type: 'module',
-        baseDirectory: '/',
-      }),
-    ).toThrow('Failed to build ES module files.');
+    expect(error).toHaveBeenCalledWith(
+      expect.stringContaining('Failed to write file.'),
+    );
   });
 });
 
```

Comment:
Looks like a bunch of the test calls to `getTransformers` still include removed parameters, e.g. `baseDirectory`